hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandleFactory.java

```diff
@@ -80,21 +80,11 @@ public static <T, I, K, O> HoodieMergeHandle<T, I, K, O> create(
       TaskContextSupplier taskContextSupplier,
       Option<BaseKeyGenerator> keyGeneratorOpt) {
     if (table.requireSortedRecords()) {
-      if (table.getMetaClient().getTableConfig().isCDCEnabled()) {
-        return new HoodieSortedMergeHandleWithChangeLog<>(writeConfig, instantTime, table, keyToNewRecords, partitionPath, fileId,
-            dataFileToBeMerged, taskContextSupplier, keyGeneratorOpt);
```

Comment:
Let's keep the method name as `create` as is and just rollback the cdc supplemental log generation for compaction, and address in the release note that MOR table CDC is not supported for release 0.13.0

Reply:
why claim the MOR table CDC is not supported?  This pr works.
And the previous method name is not easy to distinguish.

Reply:
@YannByron understand the intent, but capturing "compaction" in the name doesn't make sense. I see we capture in the java-doc that this method is meant for Compaction, that should be sufficient